06-building-a-program-start-to-finish/writing-tests/output/invertedIndexSpec.js

```diff
@@ -0,0 +1,183 @@
+      expect(index.indexed).toEqual({});
+      expect(index.showAllFilesSearch).toBeFalsy();
+      expect(index.searchAllResult).toEqual({});
+    });
+  });
+
+  describe('ValidateFile', () => {
+    it('should return false if an invalid file was uploaded', () => {
+      expect(InvertedIndex.validateFile(empty)).toBeFalsy();
+      expect(InvertedIndex.validateFile(emptyText)).toBeFalsy();
+      expect(InvertedIndex.validateFile(notArray)).toBeFalsy();
+      expect(InvertedIndex.validateFile(wrongData)).toBeFalsy();
+    });
+    it('should not return false if a valid file was uploaded', () => {
+      expect(InvertedIndex.validateFile(validFile)).not.toBeFalsy();
+      expect(InvertedIndex.validateFile(simpleValidFile)).not.toBeFalsy();
+      expect(typeof InvertedIndex.validateFile(validFile)).toEqual('object');
+    });
+    it('should return the file content if the file is valid', () => {
+      expect(InvertedIndex.validateFile(validFile)).toEqual(validFile);
+      expect(InvertedIndex.validateFile(simpleValidFile))
+      .toEqual(simpleValidFile);
+    });
+  });
+
+  describe('Tokenize', () => {
+    it('should return an array of unique words', () => {
+      expect(InvertedIndex.tokenize('tony is humble tony'))
+      .toEqual(['tony', 'is', 'humble']);
+    });
+    it('should remove invalid characters', () => {
+      expect(InvertedIndex.tokenize('$A%^n*del)a %i=s @a$w&es$om#e%'))
+      .toEqual(['andela', 'is', 'awesome']);
+    });
+    it('should return lowercase words', () => {
+      expect(InvertedIndex.tokenize('HEllO WORLD')).toEqual(['hello', 'world']);
+    });
+  });
+
+  describe('CreateIndex', () => {
+    beforeAll(() => {
+      this.filename = 'simpleValidFile.json';
+    });
+
+    it('creates an index object from the file content', () => {
+      expect(index.createIndex(simpleValidFile, this.filename)).toBeTruthy();
+      expect(typeof index.getIndex(this.filename)).toEqual('object');
+    });
+    it('returns false if file already exists', () => {
+      expect(index.createIndex(simpleValidFile, this.filename)).toBeFalsy();
+    });
+    it('correctly stores the total number of books and their titles', () => {
+      expect(index.getIndex(this.filename).bookTitles).toEqual(
+        ['Alice', 'Lord of the Rings', 'Rings']
+      );
+      expect(index.getIndex(this.filename).bookTitles.length).toBe(3);
+    });
+    it('creates the correct index object', () => {
+      expect(index.getIndex(this.filename).words.alice)
+      .toEqual([true, false, false]);
+      expect(index.getIndex(this.filename).words.falls)
+      .toEqual([true, true, false]);
+      expect(index.getIndex(this.filename).words.alliance)
+      .toEqual([false, true, true]);
+    });
+    it('correctly stores all words in the file', () => {
+      expect(index.getIndex(this.filename).allWords).toEqual(
+        ['alice', 'falls', 'into', 'an', 'unusual', 'alliance']
+      );
+    });
+  });
```

Comment:
I'd like to see you test that the index created by the `createIndex` method is accurate in this suite.